hathor/merged_mining/coordinator.py

```diff
@@ -1307,7 +1307,7 @@ async def _update_hathor_block(self, mining: IMiningChannel) -> None:
             # self.log.debug('hathor.get_block_template response', block=block, height=height)
             self.hathor_coord_job = HathorCoordJob(block, height)
             self.log.debug('new Hathor block template', height=height, weight=block.weight)
-            self.update_merged_block()
+            await self.update_merged_block()
```

Comment:
Is it a bug fix?

Reply:
Yes, it's a bug fix, although it's probably very minor.

Reply:
No tests to update?

Reply:
The updated `mypy` complained about it. No tests are affected by it. I'm not really sure how to test this though, so I'd rather leave it as is.

Reply:
I feel there's a small behavior change. Before adding the `await`, the method would keep executing regardless of the full node response. Now, it will wait for an update. What if the update fails? Will the method break?

Reply:
That makes sense, I'll change it to keep the old behavior but not have a loose future, instead I'll add it to the event loop directly. The main complaint that mypy makes is that the future was not awaited, it's technically not a mistake to not await a future, but the proper way to do it is through the event loop, because the handling of that is very implementation specific.

Reply:
I've reviewed the code. It was supposed to have an `await`, I prefer to have the small behavior change with the added `await`. If there's an exception it will be propagated to the function that called `update_merged_block`, which wouldn't before, but this isn't an issue by itself, the outer function `update_bitcoin_block` is called periodically and can fail and will recover. The only other place we call `update_merged_block` is within `update_hathor_block` (indirectly, and directly inside `_update_hathor_block`), which correctly had an `await`, and also has the behavior we should have had on `update_bitcoin_block`.

Reply:
That's great! Thanks!